tests/docker/test_docker.py

```diff
@@ -291,28 +297,103 @@ def test_import(workers, test_files):
                 ac.ArchiveFile.name == file_['name']
             )
 
+            # Check that we haven't imported types we don't want
+            if file_['type'] in [None, 'lock']:
+                assert file_query.count() == 0
+                continue
+
             assert file_query.count() == 1
             file_obj = file_query.get()
 
             # Test that it has the correct type
             assert file_obj.type.name == file_['type']
 
-            # Test that this node has a copy
-            copy_query = ar.ArchiveFileCopy.select().where(
-                ar.ArchiveFileCopy.file == file_obj,
-                ar.ArchiveFileCopy.node == node
-            )
+            if copies_on_node is not None:
+                # Test that this node has a copy
+                copy_query = ar.ArchiveFileCopy.select().where(
+                    ar.ArchiveFileCopy.file == file_obj,
+                    ar.ArchiveFileCopy.node == copies_on_node
+                )
+
+                assert copy_query.count() == 1
+                copy_obj = copy_query.get()
+
+                assert copy_obj.has_file == 'Y'
+                assert copy_obj.wants_file == 'Y'
+
+
+def _verify_files(worker):
+    """Verify the files are in place using the alpenhorn verify command.
+    """
+
+    # Run alpenhron verify and return the exit status as a string
+    output = worker['container'].exec_run(
+        "bash -c 'alpenhorn verify %s &> /dev/null; echo $?'" %
+        worker['node'].name
+    )
+
+    # Convert the output back to an exit status
+    assert not int(output)
+
+
+# ====== Test the auto_import behaviour ======
+
+def test_import(workers, test_files):
+    # Add a bunch of files onto node_0, wait for them to be picked up by the
+    # auto_import, and then verify that they all got imported to the db
+    # correctly.
 
-            assert copy_query.count() == 1
-            copy_obj = copy_query.get()
+    # Create the files
+    _make_files(test_files, workers[0]['dir'], skip_lock=True)
+
+    # Wait for the auto_import to catch them (it polls at 30s intervals)
+    time.sleep(3)
```

Comment:
The `alpenhorn.conf` used by the docker tests reduces the polling period down to 2s, so I think it's fine.